govstack_api/middleware.py

```diff
@@ -0,0 +1,89 @@
+import os
+from functools import wraps
+from types import SimpleNamespace
+from unittest import mock
+
+from graphene import Schema
+from graphene.test import Client
+
+from django.apps import apps
+from django.contrib.auth import get_user_model
+from django.http import HttpResponse
+from core.schema import Query as CoreQuery, Mutation as CoreMutation
+
+
+def authenticate_decorator(view_func):
+    @wraps(view_func)
+    def _wrapped_view(request, *args, **kwargs):
+        config = apps.get_app_config('govstack_api')
+        im_client = config.IM_CLIENT
+        im_header = request.META.get('HTTP_INFORMATION_MEDIATOR_CLIENT')
+        if im_header != im_client:
+            return HttpResponse('Unauthorized', status=401)
+        else:
+            # self.authenticate_user(request)
+            pass
+
+        return view_func(request, *args, **kwargs)
+    return _wrapped_view
```

Comment:
Is this function needed when we InformationmediatorMiddleware is in use?

Reply:
We have decorator function and middleware since in testing there is issue with accessing swagger endpoint that is used to for endpoint testing.